src/server/game/Maps/Map.cpp

```diff
@@ -2444,14 +2444,7 @@ float Map::GetWaterOrGroundLevel(uint32 phasemask, float x, float y, float z, fl
 float Map::GetHeight(float x, float y, float z, bool checkVMap /*= true*/, float maxSearchDist /*= DEFAULT_HEIGHT_SEARCH*/) const
 {
     // find raw .map surface under Z coordinates
-    float mapHeight = VMAP_INVALID_HEIGHT_VALUE;
-    if (GridMap* gmap = const_cast<Map*>(this)->GetGrid(x, y))
-    {
-        float gridHeight = gmap->getHeight(x, y);
-        if (G3D::fuzzyGe(z, gridHeight - GROUND_HEIGHT_TOLERANCE))
-            mapHeight = gridHeight;
-    }
-
+    float const mapHeight = GetGridHeight(x, y);
```

Comment:
Whats the reasoning behind this change? The point of `z` argument is to never return a height that is higher than supplied `z` (and vmgr->getHeight follows this restriction)

Reply:
I noticed some issues with blink without it... might have been because of https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/Spell.cpp#L1368 allowing shortcut paths, not sure if that's supposed to happen. Can try to remove it and change the check to only allow normal paths